api/src/models/request_info.rs

```diff
@@ -67,16 +73,40 @@ impl<'r> FromRequest<'r> for RequestInfo {
     type Error = ();
 
     async fn from_request(request: &'r Request<'_>) -> request::Outcome<Self, Self::Error> {
-        let hosts: Vec<_> = request.headers().get("host").collect();
+        // Checking if "x-forwarded-host" header is present
```

Comment:
I think the code tells the reader exactly what it does.

```suggestion
```

Most of the comments do. Please, adjust accordingly and only keep comments that document something that isn't obvious from code.